webapp/src/hooks/usePluginApi.ts

```diff
@@ -0,0 +1,36 @@
+import {useSelector, useDispatch} from 'react-redux';
+
+import {useCallback} from 'react';
+
+import {ReduxState} from 'types/common/store.d';
+
+import {setApiRequestCompletionState} from 'reducers/apiRequest';
+import {msTeamsPluginApi} from 'services';
+
+function usePluginApi() {
+    const state = useSelector((reduxState: ReduxState) => reduxState['plugins-com.mattermost.msteams-sync']);
+    const dispatch = useDispatch();
+
+    // Pass payload in POST requests only. For GET requests, there is no need to pass a payload argument
+    const makeApiRequest = useCallback(async (serviceName: PluginApiServiceName, payload: APIRequestPayload): Promise<any> => {
+        return dispatch(msTeamsPluginApi.endpoints[serviceName].initiate(payload));
+    }, [dispatch, msTeamsPluginApi.endpoints]);
+
+    const makeApiRequestWithCompletionStatus = useCallback(async (serviceName: PluginApiServiceName, payload: APIRequestPayload) => {
+        const apiRequest = await makeApiRequest(serviceName, payload);
+
+        if (apiRequest) {
+            dispatch(setApiRequestCompletionState(serviceName));
+        }
+    }, [dispatch, makeApiRequest, setApiRequestCompletionState]);
+
+    // Pass payload in POST requests only. For GET requests, there is no need to pass a payload argument
+    const getApiState = useCallback((serviceName: PluginApiServiceName, payload: APIRequestPayload) => {
```

Comment:
I'm not certain, but this feels like this `useCallback` may not be actually helping here much since it returns a new callback whenever anything in the plugin state changes, even if it's unrelated to the specific API that we want. I''m not sure if a good way to do this if we want `getApiState` to still be a callback, but if we just return `apiState` from `usePluginApi`, I think we should just be able to return `msTeamsPluginApi.endpoints[serviceName].select(payload)(state)` without wrapping it in any extra hooks